activity-monitor.js

```diff
@@ -103,13 +118,35 @@
                 if (service.user.active && service.user.action <= inactive) {
                     service.user.active = false;
                     publish(EVENT_INACTIVE);
-                    disable();
+
+                    if(service.options.disableOnInactive){
+                        disable();
+                    }else{
+                        disableIntervals();//user inactive is known, lets stop checking, for now
+                        dynamicActivity = reactivate;//hot swap method that handles document event watching
+                    }
                 }
             }, service.options.monitor * MILLISECOND);
         }
 
+        /* function that lives in memory with the intention of being swapped out */
+        function dynamicActivity(){
+            regularActivityMonitor();
+        }
+
+        /* after user inactive, this method is hot swapped as the dynamicActivity method in-which the next user activity reactivates monitors */
+        function reactivate() {
+            enableIntervals();
+            dynamicActivity = regularActivityMonitor;
+        }
+
         /* invoked on every user action */
-        function activity() {
+        function activity(){
+            dynamicActivity()
```

Comment:
What you are suggesting, would result in another condition being read during normal event triggering activity. We should minimize the logic that is run for user events. So, I disagree with you unless I'm missing details in your change in approach.... If you are just trying to keep it simple, I don't think this is the place for that. Performance for first for this package, i think.